cmd/cluster-agent/admissioncontroller/listener.go

```diff
@@ -0,0 +1,203 @@
+package admissioncontroller
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/api/admission/v1beta1"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/kubernetes/pkg/apis/core/v1"
+
+	"github.com/DataDog/datadog-agent/cmd/agent/common/jsonpatch"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var (
+	// Scheme is the schemed used for validation and mutation of resources
+	scheme = runtime.NewScheme()
+
+	// Codecs is the codec used for en/de coding
+	codecs = serializer.NewCodecFactory(scheme)
+)
+
+func init() {
+	corev1.AddToScheme(scheme)
+	admissionregistrationv1beta1.AddToScheme(scheme)
+	// defaulting with webhooks:
+	// https://github.com/kubernetes/kubernetes/issues/57982
+	v1.AddToScheme(scheme)
+}
+
+func status(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte("status OK"))
+}
+
+func serve(w http.ResponseWriter, r *http.Request) {
+	var body []byte
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+
+	log.Debug("admission controller request body: %v", body)
+
+	requestedAdmissionReview := v1beta1.AdmissionReview{}
+	responseAdmissionReview := v1beta1.AdmissionReview{}
```

Comment:
The pattern in Go is to use short names, always. The rule of thumb is:

> The greater the distance between a name's declaration and its uses,
the longer the name should be.

From the great talk [What's in a name](https://talks.golang.org/2014/names.slide#4) by Andrew Gerrand.

```suggestion
	var req, resp v1beta1.AdmissionReview
```

Short names make programming logic easier to read.